src/planscape/planning/tests/test_planningarea_views.py

```diff
@@ -0,0 +1,917 @@
+import json
+from unittest import mock
+from django.db import connection
+from django.contrib.auth.models import User
+from django.contrib.gis.geos import GEOSGeometry
+from django.urls import reverse
+from rest_framework_simplejwt.tokens import RefreshToken
+from rest_framework.test import APITransactionTestCase
+from planning.models import PlanningArea, Scenario, ScenarioResult
+from planning.tests.helpers import _create_planning_area, _create_scenario
+
+# Yes, we are pulling in an internal just for testing that a geometry write happened.
+from planning.views import _convert_polygon_to_multipolygon
+
+# TODO: Add tests to ensure that users can't have planning areas with the same
+# name in the same region, and that users can't have scenarios with the same
+# name in the same planning area.
+
+
+class CreatePlanningAreaTest(APITransactionTestCase):
+    def setUp(self):
+        self.user = User.objects.create(username="testuser")
+        self.user.set_password("12345")
+        self.user.save()
+
+        self.token = RefreshToken.for_user(self.user).access_token
+        self.geometry = {
+            "features": [
+                {
+                    "geometry": {
+                        "type": "Polygon",
+                        "coordinates": [[[1, 2], [2, 3], [3, 4], [1, 2]]],
+                    }
+                }
+            ]
+        }
+        self.multipolygon_geometry = {
+            "features": [
+                {
+                    "geometry": {
+                        "type": "MultiPolygon",
+                        "coordinates": [[[[1, 2], [2, 3], [3, 4], [1, 2]]]],
+                    }
+                }
+            ]
+        }
+        self.notes = "Inconcievable! \
+              You keep using that word. \
+              I do not think it means what you think it means."
+
+    def test_create_planning_area(self):
+        self.client.force_authenticate(self.user)
+        payload = json.dumps(
+            {
+                "name": "test plan",
+                "region_name": "Sierra Nevada",
+                "geometry": self.geometry,
+                "notes": self.notes,
+            }
+        )
+        response = self.client.post(
```

Comment:
This seems to be repeated a bunch of times. Maybe it can be abstracted to a function?